src/java.base/share/classes/jdk/internal/loader/URLClassPath.java

```diff
@@ -109,6 +110,9 @@
         // the check is not disabled).
         p = props.getProperty("jdk.net.URLClassPath.showIgnoredClassPathEntries");
         DEBUG_CP_URL_CHECK = p != null ? p.equals("true") || p.isEmpty() : false;
+
+        p = props.getProperty("jdk.net.URLClassPath.enableJarIndex");
+        ENABLE_JAR_INDEX = p != null ? p.equals("true") : false;
```

Comment:
Maybe this should use the same pattern than the other property above:

```
ENABLE_JAR_INDEX = p != null ? p.equals("true") || p.isEmpty() : false;
```

Reply:
Another question is where to document this property. We will obviously need a CSR and releases notes. I also wonder if it should be promoted into a networking property - which would make it possible to document it in the `net.properties` file. @AlanBateman what do you think?

Reply:
> Another question is where to document this property. We will obviously need a CSR and releases notes. I also wonder if it should be promoted into a networking property - which would make it possible to document it in the `net.properties` file. @AlanBateman what do you think?

jdk.net.URLClassLoader.enableJarIndex is meant to be a temporary property. We could document it in URLClassLoader as an implNote but I'm not sure that it's worth it. I re-worded the CSR to reflect the current proposal and Wang has submitted it.  I noted in the CSR that a release note is planned.

Reply:
> Maybe this should use the same pattern than the other property above:
> 
> ```
> ENABLE_JAR_INDEX = p != null ? p.equals("true") || p.isEmpty() : false;
> ```

Yes, the intention is that -Djdk.net.URLClassLoader.enableJarIndex will re-enable the feature.

Reply:
I assume the reason for specifically checking for a value of true or an empty string as you envision allowing the property to be explicitly set to false in the future?  

Otherwise we could just check if the property is set(regardless of value) to re-enable the feature.

Reply:
I can appreciate that and that is a fair point.   I guess  where I was coming from was that the property is temporary and has to be explicitly specified so ignoring the value seemed worth raising/discussing.   

 Assuming the current check is kept(which we should based on your input), it should probably be changed to ignore case.

Reply:
Given that there is a precedence in this file:

```
        p = props.getProperty("jdk.net.URLClassPath.showIgnoredClassPathEntries");
        DEBUG_CP_URL_CHECK = p != null ? p.equals("true") || p.isEmpty() : false;
```

I would tend to let the new property follow the same pattern, especially since it's temporary.

Reply:
Yes，I changed the code to follow the same pattern.

Reply:
> Given that there is a precedence in this file:
> 
> ```
>         p = props.getProperty("jdk.net.URLClassPath.showIgnoredClassPathEntries");
>         DEBUG_CP_URL_CHECK = p != null ? p.equals("true") || p.isEmpty() : false;
> ```
> 
> I would tend to let the new property follow the same pattern, especially since it's temporary.

fair enough :-)